fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/impl/DeleteResourcePersister.java

```diff
@@ -52,16 +57,68 @@ public void persist(final OCFLPersistentStorageSession session, final ResourceOp
         final var objectSession = session.findOrCreateSession(mapping.getOcflObjectId());
         log.debug("Deleting {} from {}", resourceId, mapping.getOcflObjectId());
         if (fedoraResourceRoot.equals(resourceId)) {
-            // We are at the root of the object.
-            objectSession.deleteObject();
+            // We are at the root of the object, so delete all the data files.
+            try {
+                objectSession.listHeadSubpaths().filter(p -> !isSidecarSubpath(p)).forEach(
+                        p -> deletePathWrapped(p, objectSession));
+            } catch (final PersistentStorageRuntimeException exc) {
+                // Rethrow the exception as a checked exception
+                throw new PersistentStorageException(exc.getCause().getMessage());
+            }
         } else {
             final var relativeSubPath = relativizeSubpath(fedoraResourceRoot, operation.getResourceId());
             final var ocflSubPath = resolveOCFLSubpath(fedoraResourceRoot, relativeSubPath);
-            final var headers = readHeaders(objectSession, ocflSubPath);
-            final var filePath = resolveExtensions(ocflSubPath,
-                    !Objects.equals(NON_RDF_SOURCE.toString(), headers.getInteractionModel())
-            );
-            objectSession.delete(filePath);
+            final var headers = (ResourceHeadersImpl) readHeaders(objectSession, ocflSubPath);
+            final boolean isRdf = !Objects.equals(NON_RDF_SOURCE.toString(), headers.getInteractionModel());
+            final var filePath = resolveExtensions(ocflSubPath, isRdf);
+            deletePath(filePath, objectSession, headers);
+            if (!isRdf) {
+                // Delete the description too.
+                final var descPath = resolveExtensions(ocflSubPath + "-description", true);
+                deletePath(descPath, objectSession);
+            }
+        }
+    }
+
+    /**
+     * Simple utility to delete a path's files and mark them as deleted in the headers file.
+     * @param path Path to delete
+     * @param session Session to delete the path in.
+     */
+    private void deletePath(final String path, final OCFLObjectSession session) throws PersistentStorageException{
+        // readHeaders and writeHeaders need the subpath where as delete needs the file name. So remove any extensions.
+        final var no_extension = (path.contains(".") ? path.substring(0, path.indexOf(".")) : path);
```

Comment:
Is it possible for a path to have a `.` that is unrelated to a file extension?

Reply:
unsure, we could have something that removes only specific extensions?

Reply:
I would suggest creating a JIRA to address this issue separately.